File: C++/1200. Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& a) {
        int i,m=INT_MAX;
        vector<vector<int>>d;
        sort(a.begin(),a.end());
        for(i=0;i<a.size()-1;i++){
            if(abs(a[i]-a[i+1])<m)
            m=abs(a[i]-a[i+1]);
        }
        for(i=0;i<a.size()-1;i++){
            if(abs(a[i]-a[i+1])==m)
            d.push_back({a[i],a[i+1]});
        }
        return d;
    }
};

File: JAVA/1200. Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] a) {
        int i,m=99999;
        List<List<Integer>>d=new ArrayList<>();
        Arrays.sort(a);
        for(i=0;i<a.length-1;i++){
            if(-(a[i]-a[i+1])<m)
            m=-(a[i]-a[i+1]);
        }
        for(i=0;i<a.length-1;i++){
            if(-(a[i]-a[i+1])==m)
            d.add(Arrays.asList(a[i], a[i+1]));
        }
        return d;
    }
}